clouddrift/pairs.py

```diff
@@ -0,0 +1,229 @@
+"""
+Functions to analyze pairs of contiguous data segments.
+"""
+from clouddrift import sphere
+import numpy as np
+import pandas as pd
+import xarray as xr
+from typing import Tuple, Union
+
+array_like = Union[list[float], np.ndarray[float], pd.Series, xr.DataArray]
+
+
+def pair_bounding_box_overlap(
+    lon1: array_like,
+    lat1: array_like,
+    lon2: array_like,
+    lat2: array_like,
+    tolerance: float = 0,
+) -> Tuple[np.ndarray[bool], np.ndarray[bool]]:
+    """Given two arrays of longitudes and latitudes, return boolean masks for
+    their overlapping bounding boxes.
+
+    Parameters
+    ----------
+    lon1 : array_like
+        First array of longitudes in degrees.
+    lat1 : array_like
+        First array of latitudes in degrees.
+    lon2 : array_like
+        Second array of longitudes in degrees.
+    lat2 : array_like
+        Second array of latitudes in degrees.
+    tolerance : float, optional
+        Tolerance in degrees for the overlap. If the overlap is within this
+        tolerance, the times are considered to overlap. Default is 0, or no
+        tolerance.
+
+    Returns
+    -------
+    overlap1 : np.ndarray[bool]
+        Boolean mask for the overlapping times in `lon1` and `lat1`.
+    overlap2 : np.ndarray[bool]
+        Boolean mask for the overlapping times in `lon2` and `lat2`.
+
+    Examples
+    --------
+    >>> lon1 = [0, 0, 1, 1]
```

Comment:
I don't understand what this function does and these examples a puzzling to me.

Reply:
I see that this docstring is a bad case of copy/paste (needs editing).

In short, given an input pair of lon, lat arrays, return masks that denote which elements of input arrays are inside the overlap/intersection of their bounding boxes.